### DISCUSSION TITLE
How do we enable localization of complex components?

### DISCUSSION THREAD

**Discussion by scomea** [microsoft/fast#5807] [General]
So far FAST components have generally been fairly simple and developers could typically set strings on those elements directly (button titles, aria attributes, etc...).  Now we are starting to see more complex UI's show up as part of components, for example the [date picker component](https://github.com/microsoft/fast/pull/5539) will have various flyouts that will need localized strings (even icon only buttons need accessible descriptions). 

The prototype [text editor](https://github.com/microsoft/fast/blob/a9b286b5c094be00eab302daf9bb131d22e163c9/packages/web-components/fast-foundation/src/text-editor/text-editor.ts#L62) has a generic "resources object" that templates can bind to `${x => x.resources["bulletButtonTooltip"]}`.  That was just a quick first guess at how to do this.

We should have a standard approach for how we implement this, so what should that approach be?

Thoughts?

**Comment by KingOfTac** [comment]
One Idea I've had, at least for static strings is to use the component definition to pass localized strings and icons to components when they get registered with the design system.

**Reply by scomea** [reply]
Would that make switching languages a bigger reset than it should be?   As a Canadian I'm used to seeing french/english toggles everywhere.

**Reply by KingOfTac** [reply]
Interesting, I hadn't thought about toggling back and forth. I was mostly thinking of detecting the browser language and setting all the strings once.

**Reply by scomea** [reply]
Also, there is probably a notion of default strings for a component and being able to override on a per-instance basis too.
